UnityAds/src/main/java/com/mopub/mobileads/UnityVideoAd.java

```diff
@@ -0,0 +1,153 @@
+    protected int missedImpressionOrdinal;
+
+    protected abstract IUnityAdsShowListener getIUnityAdsShowListener();
+
+    @Override
+    protected void onInvalidate() { }
+
+    @Override
+    protected void load(@NonNull Context context, @NonNull AdData adData) {
+        super.load(context, adData);
+        if (failIfPlacementIdNull()) {
+            return;
+        }
+
+        MoPubLog.log(getAdNetworkId(), LOAD_ATTEMPTED, ADAPTER_NAME);
+        UnityAdsLoadOptions loadOptions = new UnityAdsLoadOptions();
+
+        String markup = adData.getExtras().get(DataKeys.ADM_KEY);
+        if (markup != null) {
+            loadOptions.setAdMarkup(markup);
+            mObjectId = UUID.randomUUID().toString();
+            loadOptions.setObjectId(mObjectId);
+        }
+        UnityAds.load(mPlacementId, loadOptions, mUnityLoadListener);
+    }
+
+    /**
+     * IUnityAdsLoadListener instance. Contains ad load success and fail logic.
+     */
+    private IUnityAdsLoadListener mUnityLoadListener = new IUnityAdsLoadListener() {
+        @Override
+        public void onUnityAdsAdLoaded(String placementId) {
+            mPlacementId = placementId;
+            MoPubLog.log(CUSTOM, ADAPTER_NAME, LOG_LOAD_SUCCESS + placementId);
+            MoPubLog.log(LOAD_SUCCESS, ADAPTER_NAME);
+
+            if (mLoadListener != null) {
+                mLoadListener.onAdLoaded();
+            }
+        }
+
+        @Override
+        public void onUnityAdsFailedToLoad(String placementId, UnityAds.UnityAdsLoadError error, String message) {
+            mPlacementId = placementId;
+            MoPubLog.log(CUSTOM, ADAPTER_NAME, LOG_LOAD_FAILED + placementId);
+            MoPubLog.log(LOAD_FAILED, ADAPTER_NAME, MoPubErrorCode.NETWORK_NO_FILL.getIntCode(), MoPubErrorCode.NETWORK_NO_FILL);
+
+            if (mLoadListener != null) {
+                mLoadListener.onAdLoadFailed(MoPubErrorCode.NETWORK_NO_FILL);
+            }
+        }
+    };
+
+    @Override
+    public void show() {
+        MoPubLog.log(SHOW_ATTEMPTED, ADAPTER_NAME);
+        MediationMetaData mediationMetaData = new MediationMetaData(mActivity);
+
+        if (mActivity == null || mPlacementId == null) {
+            if (mActivity == null) {
+                MoPubLog.log(CUSTOM, ADAPTER_NAME, LOG_SHOW_ACTIVITY_NULL);
+            }
+            if (mPlacementId == null) {
+                MoPubLog.log(CUSTOM, ADAPTER_NAME, LOG_PLACEMENT_ID_MISSING);
+            }
+            MoPubLog.log(SHOW_FAILED, ADAPTER_NAME, MoPubErrorCode.VIDEO_PLAYBACK_ERROR.getIntCode(), MoPubErrorCode.VIDEO_PLAYBACK_ERROR);
+
+            mediationMetaData.setMissedImpressionOrdinal(++missedImpressionOrdinal);
+            mediationMetaData.commit();
+
+            if (mInteractionListener != null) {
+                mInteractionListener.onAdFailed(MoPubErrorCode.VIDEO_PLAYBACK_ERROR);
+            }
+            return;
+        }
+
+        mediationMetaData.setOrdinal(++impressionOrdinal);
```

Comment:
I believe mediation metadata is no longer needed. cc @andrewkonecny to confirm

Reply:
Mediation Ordinal (a.k.a the amount of impressions shown in the mediation SDK) is still required. It's used for Comet for better targeting purposes.

The previous conversation was around the `missedImpressionOrdinal`, which is technically not required, but we should keep it for the time being, since it's direct feedback that we missed an opportunity to show an Ad due to misalignment between state of Mediation SDK and Ads SDK.

Reply:
hm, it was never implemented on ios mopub adapter. I'll add to my pr as well then